src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2906,7 +2910,11 @@ private void onApplicationWindow(
                 if (subscription != null &&
                         !subscription.hasSubscribeCompleted(subackIndex) && hasSubscribeCapability(capabilities))
                 {
-                    subscription.onSubscribeSucceeded(traceId, authorization, packetId, subackIndex);
+                    final boolean completed = subscription.onSubscribeSucceeded(traceId, authorization, packetId, subackIndex);
+                    if (completed)
+                    {
+                        doApplicationWindow(traceId, authorization);
+                    }
```

Comment:
Shouldn't this conditional sending of `window` be done in `onApplicationBegin` instead?

Also, you need to handle the case where `begin` arrived for multiple reply streams, and the first one couldn't send `window` because the `subscription` had not yet been `acknowledged`, then the last reply stream comes in, allowing the `subscription` to be `acknowledged` so all application streams for the `subscription` should now be notified to send `window` if necessary otherwise they would be stalled.